.github/workflows/main.yml

```diff
@@ -14,96 +14,111 @@ on:
+    with:
+      branch_name: ${{ github.ref }}
 
   license-check:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
-
-      - name: reuse lint
-        uses: fsfe/reuse-action@v5
+    uses: gatici/.github/.github/workflows/license-check.yml@add-workflows
+    with:
+      branch_name: ${{ github.ref }}
 
-  fossa-check:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
-
-      - name: FOSSA scan
-        uses: fossa-contrib/fossa-action@v3
-        with:
-          fossa-api-key: 0c3bbcdf20e157bbd487dae173751b28
+  fossa-scan:
+    uses: gatici/.github/.github/workflows/fossa-scan.yml@add-workflows
+    with:
+      branch_name: ${{ github.ref }}
 
   unit-tests:
-    runs-on: ubuntu-latest
-    steps:
-      - uses: actions/checkout@v4
+    uses: gatici/.github/.github/workflows/unit-test.yml@add-workflows
+    with:
+      branch_name: ${{ github.ref }}
 
-      - uses: actions/setup-go@v5
-        with:
-          go-version-file: 'go.mod'
+  e2e-tests:
+    runs-on: ubuntu-latest
 
-      - name: Unit tests
-        run: go test ./...
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Update and install dependencies
+        run: |
+          sudo apt update
+          sudo apt install -y build-essential git iptables libsctp1 openssh-server pipx
+          pipx install --include-deps 'ansible<2.16'
+          pipx ensurepath
+
+      - name: Install Ansible Galaxy collections
+        run: |
+          ansible-galaxy collection install kubernetes.core ansible.utils community.docker
+
+      - name: Generate SSH Key
+        run: |
+          ssh-keygen -t rsa -b 2048 -f ~/.ssh/id_rsa -N "" -q
+          cat ~/.ssh/id_rsa.pub >> ~/.ssh/authorized_keys
+
+      - name: Clone aether-onramp repository
+        run: |
+          git clone --recursive https://github.com/opennetworkinglab/aether-onramp.git
+          cd aether-onramp
+
+      - name: Update files
+        run: |
+          IP_ADDR=$(ip a show eth0 | grep -oP '(?<=inet\s)\d+(\.\d+){3}')
+          echo "Extracted IP: ${IP_ADDR}"
+          ls -l ~/.ssh/id_rsa
+          id
+          free -m
+          lscpu
+          echo "[all]" > aether-onramp/hosts.ini
+          echo "node1 ansible_host=${IP_ADDR} ansible_user=runner ansible_ssh_private_key_file=~/.ssh/id_rsa" >> aether-onramp/hosts.ini
+          echo "" >> aether-onramp/hosts.ini
+          echo "[master_nodes]" >> aether-onramp/hosts.ini
+          echo "node1" >> aether-onramp/hosts.ini
+          sed -i "s/ens18/eth0/" aether-onramp/vars/main.yml
+          sed -i "s/10.76.28.113/${IP_ADDR}/" aether-onramp/vars/main.yml
+
+      - name: Test aether-pingall
+        working-directory: aether-onramp
+        run: make aether-pingall
+
+      - name: Install aether-k8s
+        working-directory: aether-onramp
+        run: |
+          make aether-k8s-install
```

Comment:
I see this action is failing because the K8s cluster is not ready. If we can install a "lightweight" K8s cluster using a GHA or in another way, we could skip this step (using OnRamp to deploy K8s). What do you think?